cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -92,7 +97,13 @@ public BasicMapReduceContext(Program program,
     } else {
       this.userMetrics = null;
     }
-    this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    String adapterName = getAdapterName(runtimeArguments);
```

Comment:
Is the `runtimeArguments` the user runtime arguments? I think it should come from `ProgramOptions.getArguments`, but not `ProgramOptions.getUserArguments`.